packages/react/src/components/table/table-header.tsx

```diff
@@ -63,19 +76,22 @@ const StyledSortButtonIcon = styled(SortButtonIcon)`
     margin-left: var(--spacing-1x);
 `;
 
-function getHeading<TData extends object, TValue>(
-    header: CustomHeader<TData, TValue>,
-): ReactElement {
-    const currentSort = header.column.getIsSorted();
-    let sortState: SortState = 'none';
-
-    if (currentSort === 'asc') {
-        sortState = 'ascending';
-    }
-    if (currentSort === 'desc') {
-        sortState = 'descending';
+function getSortState(currentSort: false | SortDirection): SortState {
+    if (!currentSort) {
+        return 'none';
     }
 
+    return {
+        asc: 'ascending',
+        desc: 'descending',
+    }[currentSort] as SortState;
```

Comment:
Ce serait peut-être plus direct comme association
```suggestion
    switch (currentSort) {
        case 'asc':
            return 'ascending';
        case 'desc':
            return 'descending';
        default:
            return 'none';
    }
```